solutions/simple/variables.tf

```diff
@@ -0,0 +1,399 @@
+  }
+}
+
+variable "cos_bucket_class" {
+  type        = string
+  default     = "standard"
+  description = "The storage class of the newly provisioned Cloud Object Storage bucket. Specify one of the following values for the storage class: `standard`, `vault`, `cold`, `smart` (default), or `onerate_active`."
+  validation {
+    condition     = contains(["standard", "vault", "cold", "smart", "onerate_active"], var.cos_bucket_class)
+    error_message = "Specify one of the following values for the `cos_bucket_class`: `standard`, `vault`, `cold`, `smart`, or `onerate_active`."
+  }
+}
+
+###############################################################################################################
+# KMS
+###############################################################################################################
+
+variable "existing_kms_instance_crn" {
+  type        = string
+  default     = null
+  description = "The CRN of the existing key management service (KMS) that is used to create keys for encrypting the Cloud Object Storage bucket."
+}
+
+variable "kms_endpoint_type" {
+  type        = string
+  description = "The type of endpoint to use for communicating with the Key Protect instance. Possible values: `public`, `private`. Applies only if `existing_cos_kms_key_crn` is not specified."
+  default     = "public"
+  validation {
+    condition     = can(regex("public|private", var.kms_endpoint_type))
+    error_message = "Valid values for the `kms_endpoint_type_value` are `public` or `private`."
+  }
+}
+
+variable "kms_key_ring_name" {
+  type        = string
+  default     = "cos-key-ring"
+  description = "The name of the key ring to create for the Cloud Object Storage bucket key. If an existing key is used, this variable is not required. If the prefix input variable is passed, the name of the key ring is prefixed to the value in the `<prefix>-value` format."
+}
+
+variable "kms_key_name" {
+  type        = string
+  default     = "cos-key"
+  description = "The name of the key to create for the Cloud Object Storage bucket. If an existing key is used, this variable is not required. If the prefix input variable is passed, the name of the key is prefixed to the value in the `<prefix>-value` format."
+}
+
+##############################################################################
+# Optional VPC Variables
+##############################################################################
+
+variable "default_network_acl_name" {
+  description = "OPTIONAL - Name of the Default ACL. If null, a name will be automatically generated"
+  type        = string
+  default     = null
+}
+
+variable "default_security_group_name" {
+  description = "OPTIONAL - Name of the Default Security Group. If null, a name will be automatically generated"
+  type        = string
+  default     = null
+}
+
+variable "default_routing_table_name" {
+  description = "OPTIONAL - Name of the Default Routing Table. If null, a name will be automatically generated"
```

Comment:
The guidance from docs team is not to add `OPTIONAL` in the variable description. In projects it will show in the optional tab, so no need to duplicate that

Reply:
done